### PR TITLE
Use mathrm instead of mbox for text literals

### PR DESCRIPTION
**Pull request by physikerwelt**
Currently literals (i.e. literals_1) are surrounded by
mbox. For example \AA is transformed to \mbox{\AA}.
In this change we use mathrm instead mbox to surround the text literal
literal since this is supported by MathJax. For example
\AA is now transformed to \mathrm{\AA}.

### CODE REVIEW

**Comment by physikerwelt** [comment]
@wikimedia/math what do you think?

**Comment by cscott** [comment]
I don't think this is the right solution at all, see discussion at https://github.com/mathjax/MathJax/issues/1236#issuecomment-130773479 and a better solution presented (overriding `InternalText`) at https://github.com/mathjax/MathJax/issues/1236#issuecomment-129222235

**Comment by physikerwelt** [comment]
I don't think that's the RIGHT solution either. However, I think it's the most pragmatic solution. I guess overwriting InternalText might cause problem at a larger scale. Note that this change only affects a very limited and well defined number of commands.
Furhtermore, I don't see who (which developer) is willing to touch the InternalText method. I'm not a fan of deviations from upstream MathJax since that has caused a lot of trouble in the past.

**Comment by cscott** [comment]
I don't agree that it's the most pragmatic solution, I think that
pretending text mode stuff is really in math mode is causing problems, for
example with the fonts for \AA etc.  Switching \mbox to \mathrm breaks all
the users of texvcjs who are *actually using TeX* to render the output.  So
that's not acceptable, except under some sort of "broken behavior for
MathJaX" flag.

**Comment by physikerwelt** [comment]
@cscott, @ClaudeZou any ideas how to implement that MathJax flag

**Comment by physikerwelt** [comment]
@wikimedia/math I implemented the MathJax output switch. Please have a look at the PR. Thanks

File: lib/parser.pegjs [hunk 1]
```diff
@@ -253,12 +253,18 @@ LITERAL
    { return ast.RenderT.TEX_ONLY("\\operatorname {" + f.slice(1) + "}" + c); }
  / f:generic_func &{ return tu.other_literals1[f]; } _ // from Texutil.find(...)
    { return ast.RenderT.TEX_ONLY(f + " "); }
+ / f:generic_func &{ return options.mathjax && tu.other_literals2[f]; } _ // from Texutil.find(...)
+   { return ast.RenderT.TEX_ONLY("\\mathrm {" + f + "} "); }
```
Comment (d00rman) [inline lib/parser.pegjs]: Why the extra space between `/mathrm` and `{` ?`mbox` (below) doesn't add it.

**Comment by d00rman** [comment]
@cscott AFAICT, this PR proposes to replace `mbox` with `mathrm` only when the `mathjax` option is set, and as such I find it rather conservative and backwards-compatible.

I'd probably name it `usemathrm` or something similar, though, since it is really about using `mathrm` instead of `mbox`.

File: lib/parser.pegjs (hunk 1 above)
Comment (physikerwelt) [inline lib/parser.pegjs]: That's how texvcv(js) does the spacing.
```
./texvccheck '\mbox{a+b}'
+{\mbox{a+b}}
~/math/texvccheck$ ./texvccheck '\mathrm{a+b}'
+{\mathrm  {a+b}}
~/math/texvccheck$ ./texvccheck '\mbox {a+b}'
+{\mbox{a+b}}
~/math/texvccheck$ ./texvccheck '\mathrm {a+b}'
+{\mathrm  {a+b}}
```

**Comment by physikerwelt** [comment]
I prefer mathjax over usemathrm for the option name, since it's the first MathJax specfic hack... and probably not the last one.

**Comment by d00rman** [comment]
> I prefer mathjax over usemathrm for the option name, since it's the first MathJax specfic hack... and probably not the last one.

The hack is needed/used by MathJax, but that doesn't mean other clients wouldn't find it useful. My problem with the name is that it is really obscure and it's not intuitive at all. That said, we could introduce a `mathjax` option that turns on any MathJax-specific options, but the options themselves should have clear names.

**Comment by physikerwelt** [comment]
@wikimedia/math OK. How should we proceed here. I think it would be nice to have that fixed before mathoid 0.2.9 is released.

File: bin/texvcjs [hunk 2]
```diff
@@ -12,6 +12,7 @@ program
     .usage('[options] <tex input>')
     .option('--rebuild', 'Rebuild PEGjs grammar before parsing')
     .option('-v, --verbose', 'Show verbose error information')
+    .option('--usemathrm ', 'Use \\mathrm instead of \\mbox to escape some text literals')
```
Comment (d00rman) [inline bin/texvcjs]: Why the extra space at the end of the option?

Comment (physikerwelt) [inline bin/texvcjs]: That's not an easy question ... probably because I accidently hit the space-bar ;-)

**Comment by d00rman** [comment]
LGTM, but more thing - this change probably deserves a version bump

**Comment by physikerwelt** [comment]
@d00rman I support the version bump... maybe together with #8 
But @cscott manages the versions, so it's up to him

**Pull request merged**